src/keyring/KeyringController.ts

```diff
@@ -341,24 +362,27 @@ export class KeyringController extends BaseController<KeyringConfig, KeyringStat
    *
    * @param messageParams - TypedMessageParams object to sign
    * @param version - Compatibility version EIP712
+   * @throws Will throw when passed an unrecognized version
    * @returns - Promise resolving to a signed message string or an error if any
    */
-  async signTypedMessage(messageParams: TypedMessageParams, version: string) {
+  async signTypedMessage(messageParams: TypedMessageParams, version: SignTypedDataVersion) {
     try {
       const address = normalizeAddress(messageParams.from);
       const { password } = privates.get(this).keyring;
       const privateKey = await this.exportAccount(password, address);
       const privateKeyBuffer = ethUtil.toBuffer(ethUtil.addHexPrefix(privateKey));
       switch (version) {
-        case 'V1':
+        case SignTypedDataVersion.V1:
           // signTypedDataLegacy will throw if the data is invalid.
           return signTypedDataLegacy(privateKeyBuffer, { data: messageParams.data as any });
-        case 'V3':
+        case SignTypedDataVersion.V3:
           return signTypedData(privateKeyBuffer, { data: JSON.parse(messageParams.data as string) });
-        case 'V4':
+        case SignTypedDataVersion.V4:
           return signTypedData_v4(privateKeyBuffer, {
             data: JSON.parse(messageParams.data as string),
           });
+        default:
+          throw new Error(`Unexpected signTypedMessage version: '${version}'`);
```

Comment:
This is a functional change. Previously it would return `undefined` if given an invalid version.